src/cli.rs

```diff
@@ -0,0 +1,166 @@
+    /// NOTE - You may need to add '--color=always' to your subcommand in case you want to output
+    /// colour as well
+    #[arg(short, long)]
+    pub after_build: Option<String>,
+
+    /// Create source_dirs.json
+    ///
+    /// This creates a source_dirs.json file at the root of the monorepo, which is needed when you
+    /// want to use Reanalyze
+    #[arg(short, long, default_value_t = false, num_args = 0..=1)]
+    pub create_sourcedirs: bool,
+
+    /// Build development dependencies
+    ///
+    /// This is the flag to also compile development dependencies
+    /// It's important to know that we currently do not discern between project src, and
+    /// dependencies. So enabling this flag will enable building _all_ development dependencies of
+    /// _all_ packages
+    #[arg(long, default_value_t = false, num_args = 0..=1)]
+    pub dev: bool,
+
+    /// Disable timing on the output
+    #[arg(short, long, default_value_t = false, num_args = 0..=1)]
+    pub no_timing: bool,
+
+    /// Path to bsc
+    #[arg(long)]
+    pub bsc_path: Option<String>,
+}
+
+#[derive(Args, Debug)]
+pub struct WatchArgs {
+    /// Filter files by regex
+    ///
+    /// Filter allows for a regex to be supplied which will filter the files to be compiled. For
+    /// instance, to filter out test files for compilation while doing feature work.
+    #[arg(short, long)]
+    pub filter: Option<String>,
+
+    /// Action after build
+    ///
+    /// This allows one to pass an additional command to the watcher, which allows it to run when
+    /// finished. For instance, to play a sound when done compiling, or to run a test suite.
+    /// NOTE - You may need to add '--color=always' to your subcommand in case you want to output
+    /// colour as well
+    #[arg(short, long)]
+    pub after_build: Option<String>,
+
+    /// Create source_dirs.json
+    ///
+    /// This creates a source_dirs.json file at the root of the monorepo, which is needed when you
+    /// want to use Reanalyze
+    #[arg(short, long, default_value_t = false, num_args = 0..=1)]
+    pub create_sourcedirs: bool,
+
+    /// Build development dependencies
+    ///
+    /// This is the flag to also compile development dependencies
+    /// It's important to know that we currently do not discern between project src, and
+    /// dependencies. So enabling this flag will enable building _all_ development dependencies of
+    /// _all_ packages
+    #[arg(long, default_value_t = false, num_args = 0..=1)]
+    pub dev: bool,
+
+    /// Path to bsc
+    #[arg(long)]
+    pub bsc_path: Option<String>,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Command {
+    /// Build using Rewatch
+    Build(BuildArgs),
+    /// Build, then start a watcher
+    Watch(WatchArgs),
+    /// Clean the build artifacts
+    Clean {
+        /// Path to bsc
+        #[arg(long)]
+        bsc_path: Option<String>,
+    },
+    /// Alias to `legacy format`.
```

Comment:
@Bushuo when you mention legacy format, does that mean there will be a new format as well? Something that can deal with formatting mono repos as well?